### PR TITLE
Add endpoint to enable automatic sending of transaction - Closes #5880

### LINKED ISSUE #5880: Create API endpoint for submitting password to PoM plugin
### Description

Endpoint required to submit password for signing the PoM transaction (Refer to forger-plugin).

1. Encrypted passphrase will be retrieved from config.
2. Using the password from the endpoint, the passphrase is decrypted and stored in memory. 

### Motivation

In order for PoM plugin to properly sign and submit a PoM transaction, it requires an API endpoint for user to submit password to decrypt the passphrase. 

### Acceptance Criteria

- Test endpoint
- Write relevant tests

### PR DESCRIPTION
**Pull request by shuse2**
### What was the problem?

This PR resolves #5880 

### How was it solved?

- Add endpoint to enable automatic transaction sending

### How was it tested?

- Add functional tests

### CODE REVIEW

File: framework-plugins/lisk-framework-report-misbehavior-plugin/src/controllers/auth.ts [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+		const changedState = body.enable ? 'enabled' : 'disabled';
+		res.status(200).json({
+			meta: {},
+			data: {
+				message: `Successfully ${changedState} the reporting of misbehavior.`,
+			},
+		});
+	} catch (error) {
+		res.status(400).send({
+			errors: [{ message: (error as Error).message ?? 'Pasword given is not valid.' }],
```
Comment (ManuGowda) [inline framework-plugins/lisk-framework-report-misbehavior-plugin/src/controllers/auth.ts:78]:
```suggestion
			errors: [{ message: (error as Error).message ?? 'Password given is not valid.' }],
```

File: framework-plugins/lisk-framework-report-misbehavior-plugin/src/defaults/default_config.ts [hunk 2]
```diff
@@ -55,7 +55,7 @@ export const defaultConfig = {
 		dataPath: {
 			type: 'string',
 			format: 'path',
-			example: '~/.lisk/forger',
+			example: '~/.lisk/report-misbehavior',
 			description:
 				'The data path for storing forging related information captured from application.',
```
Comment (ManuGowda) [inline framework-plugins/lisk-framework-report-misbehavior-plugin/src/defaults/default_config.ts:60]:
```suggestion
				'The data path for storing misbehavior related information captured from application.',
```

Comment (ManuGowda) [reply framework-plugins/lisk-framework-report-misbehavior-plugin/src/defaults/default_config.ts:60]: other places in this file has usage of `forging`

File: framework-plugins/lisk-framework-report-misbehavior-plugin/test/functional/auth.spec.ts [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+import {
+	createApplication,
+	closeApplication,
+	waitNBlocks,
+	getURL,
+	getReportMisbehaviorPlugin,
+} from '../utils/application';
+import { defaultAccount } from '../fixtures/devnet';
+
+describe('api/forging', () => {
```
Comment (ManuGowda) [inline framework-plugins/lisk-framework-report-misbehavior-plugin/test/functional/auth.spec.ts:25]:
```suggestion
describe('Auth', () => {
```

File: framework-plugins/lisk-framework-report-misbehavior-plugin/test/utils/application.ts [hunk 4]
```diff
@@ -0,0 +1,144 @@
...
+		});
+	});
+};
+
+export const createApplication = async (
+	label: string,
+	options: {
+		consoleLogLevel?: string;
+		clearDB?: boolean;
+		appConfig?: { plugins: { forger: object } };
```
Comment (ManuGowda) [inline framework-plugins/lisk-framework-report-misbehavior-plugin/test/utils/application.ts:49]:
```suggestion
		appConfig?: { plugins: { reportMisbehavior: object } };
```

File: framework-plugins/lisk-framework-report-misbehavior-plugin/test/utils/application.ts [hunk 5]
```diff
@@ -0,0 +1,144 @@
...
+	} = {
+		clearDB: true,
+		consoleLogLevel: 'fatal',
+		appConfig: { plugins: { forger: {} } },
```
Comment (ManuGowda) [inline framework-plugins/lisk-framework-report-misbehavior-plugin/test/utils/application.ts:53]:
```suggestion
		appConfig: { plugins: { reportMisbehavior: {} } },
```

**Review by ManuGowda — APPROVED** [review]

File: framework-plugins/lisk-framework-report-misbehavior-plugin/src/controllers/auth.ts [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+		},
+	},
+};
+
+interface RequestBody {
+	readonly password: string;
+	readonly enable: boolean;
+}
+
+export const auth = (options: Options, states: States) => async (
```
Comment (mitsuaki-u) [inline framework-plugins/lisk-framework-report-misbehavior-plugin/src/controllers/auth.ts:43]: `state` can be plural

**Review by mitsuaki-u — APPROVED** [review]
LGTM one minor comment

**Pull request merged**